File: lesson06/app/src/main/java/br/odb/nehe/lesson06/GL2JNIActivity.java
package br.odb.nehe.lesson06;

import android.app.Activity;
import android.content.res.AssetManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import java.io.IOException;

public class GL2JNIActivity extends Activity {

    private GL2JNIView mView;
    private boolean running = false;
    private static AssetManager assets;


    @Override
    protected void onPause() {
        super.onPause();

        running = false;
        mView.onPause();
        GL2JNILib.onDestroy();
    }

    @Override
    protected void onResume() {
        super.onResume();

        mView = new GL2JNIView(getApplication());
        setContentView(mView);

        assets = getAssets();
        GL2JNILib.onCreate(assets);


        Bitmap bitmap;
        try {
            bitmap = BitmapFactory.decodeStream(assets.open("texture.png"));
            GL2JNILib.setTexture(bitmap);
        } catch (IOException ignored) {
        }

        mView.onResume();

        new Thread(new Runnable() {
            @Override
            public void run() {
                running = true;
                while (running) {
                    try {
                        Thread.sleep(20);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                    GL2JNILib.tick();
                }
            }
        }).start();
    }
}

File: lesson06/app/src/main/java/br/odb/nehe/lesson06/GL2JNIActivity.kt
package br.odb.nehe.lesson06

import android.app.Activity
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import java.io.IOException

class GL2JNIActivity : Activity() {
	private var mView: GL2JNIView? = null
	private var running = false
	override fun onPause() {
		super.onPause()
		running = false
		mView!!.onPause()
		GL2JNILib.onDestroy()
	}

	override fun onResume() {
		super.onResume()
		mView = GL2JNIView(application)
		setContentView(mView)
		GL2JNILib.onCreate(assets)
		val bitmap: Bitmap
		try {
			bitmap = BitmapFactory.decodeStream(assets.open("texture.png"))
			GL2JNILib.setTexture(bitmap)
		} catch (ignored: IOException) {
		}
		mView!!.onResume()
		Thread {
			running = true
			while (running) {
				try {
					Thread.sleep(20)
				} catch (e: InterruptedException) {
					e.printStackTrace()
				}
				GL2JNILib.tick()
			}
		}.start()
	}
}